### PR TITLE
[SPARK-16374] [SQL] Remove Alias from MetastoreRelation and SimpleCatalogRelation

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

Different from the other leaf nodes, `MetastoreRelation` and `SimpleCatalogRelation` have a pre-defined `alias`, which is used to change the qualifier of the node. However, based on the existing alias handling, alias should be put in `SubqueryAlias`. 

This PR is to separate alias handling from `MetastoreRelation` and `SimpleCatalogRelation` to make it consistent with the other nodes. It simplifies the signature and conversion to a `BaseRelation`.

For example, below is an example query for `MetastoreRelation`,  which is converted to a `LogicalRelation`:

``` SQL
SELECT tmp.a + 1 FROM test_parquet_ctas tmp WHERE tmp.a > 2
```

Before changes, the analyzed plan is

```
== Analyzed Logical Plan ==
(a + 1): int
Project [(a#951 + 1) AS (a + 1)#952]
+- Filter (a#951 > 2)
   +- SubqueryAlias tmp
      +- Relation[a#951] parquet
```

After changes, the analyzed plan becomes

```
== Analyzed Logical Plan ==
(a + 1): int
Project [(a#951 + 1) AS (a + 1)#952]
+- Filter (a#951 > 2)
   +- SubqueryAlias tmp
      +- SubqueryAlias test_parquet_ctas
         +- Relation[a#951] parquet
```

**Note: the optimized plans are the same.**

For `SimpleCatalogRelation`, the existing code always generates two Subqueries. Thus, no change is needed.
#### How was this patch tested?

Added test cases.

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala [hunk 1]
```diff
@@ -375,7 +377,7 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
         // Read path
         case relation: MetastoreRelation if shouldConvertMetastoreParquet(relation) =>
           val parquetRelation = convertToParquetRelation(relation)
-          SubqueryAlias(relation.alias.getOrElse(relation.tableName), parquetRelation)
+          SubqueryAlias(relation.tableName, parquetRelation)
```
Comment (gatorsmile) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: The added test case is for verifying the code changes here.

**Comment by gatorsmile** [comment]
cc @cloud-fan Do you think this removal is OK? Thank you!

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala [hunk 2]
```diff
@@ -716,6 +717,39 @@ class ParquetSourceSuite extends ParquetPartitioningTest {
     }
   }
 
+  test("Alias used in converted data source tables") {
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala]: Do we really need a test for it? This PR looks like a cleanup to me, and making one `Subquery` into two is not our target, making the code cleaner is.

Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala]: Yeah, I can remove it. 

You are right. That example is just for showing the outcome of the code changes. Thanks!

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**